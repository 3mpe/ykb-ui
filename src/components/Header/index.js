import React from "react";

function Header() {
    return (
        <header className="header">
            <h3>UI Development Challenge</h3>
        </header>
    );
}

export default Header;
