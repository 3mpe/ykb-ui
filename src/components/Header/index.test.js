import React from "react";
import {render} from "@testing-library/react";

import Header from "./index";

describe("<Header />", () => {
  it("should render the header", () => {
    const { container } = render(<Header title="UI Development Challenge"/>);

    const selectedHeader = container.querySelectorAll('.header');
    const headerLength = selectedHeader.length;
    expect(headerLength).toBe(1);

  });

  it("should render the header text prop", () => {
    const { container } = render(<Header title="UI Development Challenge"/>);
    const selectedHeader = container.querySelectorAll('.header');
    const selectedHeaderText = selectedHeader[0].textContent;

    expect(selectedHeaderText).toBe("UI Development Challenge");

  });

  it('should toMatchSnapshot', function () {
    const { container } = render(<Header title="UI Development Challenge"/>);
    expect(container.firstChild).toMatchSnapshot()
  });
});