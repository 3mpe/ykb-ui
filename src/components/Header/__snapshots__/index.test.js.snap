// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should toMatchSnapshot 1`] = `
<header
  class="header"
>
  <h3>
    UI Development Challenge
  </h3>
</header>
`;
